addons/payment/utils.py

```diff
@@ -110,7 +110,7 @@ def to_minor_currency_units(major_amount, currency, arbitrary_decimal_number=Non
     else:
         currency.ensure_one()
         decimal_number = currency.decimal_places
-    return int(float_round(major_amount, decimal_number) * (10**decimal_number))
+    return int(float_round(major_amount * (10**decimal_number), decimal_number))
```

Comment:
Since you already decupled `major amount`, the `precision_digits` must be `0` to get the right minor amount. The numbers after the comma are no longer relevant to this currency.

```suggestion
    return int(float_round(major_amount * (10**decimal_number), precision_digits=0))
```

If you let the `precision_digits` to `decimal_number`, there will still be issues:
```
In [41]: odoo.tools.float_round(2.0499 * 10**2, precision_digits=2)
Out[41]: 204.99
```